app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivity.kt

```diff
@@ -275,11 +290,53 @@ class DeviceShieldTrackerActivity :
     }
 
     private fun launchAlwaysOnPromotionDialog() {
-        val dialog = AlwaysOnAlertDialogFragment.newInstance { viewModel.onViewEvent(ViewEvent.PromoteAlwaysOnOpenSettings) }
-        dialog.show(
-            supportFragmentManager,
-            TAG_APPTP_PROMOTE_ALWAYS_ON_DIALOG
-        )
+        if (this::alwaysOnDialog.isInitialized) {
```

Comment:
don't do this to me, check with the `FragmentManager` if the `Fragment` exists using the `TAG`

Reply:
oohh, yeah, I think it's better ;)